src/components/Presenter.js

```diff
@@ -0,0 +1,358 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import './Presenter.css';
+
+const PresenterStatus = {
+  INIT: 0,
+  LOADED: 1,
+  PLAYING: 2,
+  PAUSED: 3,
+  STOPPED: 4,
+};
+
+const ItemStatus = {
+  INIT: 0,
+  LOADED: 1,
+  ENTERING: 2,
+  ENTERED: 3,
+  LEAVING: 4,
+  LEFT: 5,
+};
+
+export const PresenterDirection = {
+  RIGHT: 2,
+  LEFT: 3,
+};
+
+const TransitionHorizontalSlide = {
+  // STYLE TO DISPLAY THE SLIDE
+  normal: {
+    opacity: 1,
+    left: 0,
+    right: 0,
+  },
+
+  // STYLE TO START SLIDE
+  enter: {
+    opacity: 0,
+    left: '-100%',
+    right: '100%',
+  },
+
+  // STYLE TO FINISH SLIDE
+  left: {
+    opacity: 0,
+    left: '100%',
+    right: '-100%',
+  },
+};
+
+class PresenterItem extends React.Component {
```

Comment:
Is better this to sit on its own `PresenterItem` js file as this can be quite simple to test even

Reply:
Ok